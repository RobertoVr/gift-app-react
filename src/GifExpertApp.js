import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories);

    return (
        <div>
            <h2>GifExpert</h2>
            <AddCategory setCategories={setCategories} />
            <hr></hr>
            <ol>
                {
                    categories.map(cat => (
                        <GifGrid key={cat} category={cat} />
                    ))
                }
            </ol>
        </div>
    )
}
