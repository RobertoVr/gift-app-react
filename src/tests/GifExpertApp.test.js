import React from 'react';
import '@testing-library/jest-dom';
import { GifExpertApp } from '../GifExpertApp';
import { shallow } from 'enzyme';

describe('Probando componente <GifExpertApp />', () => {
    test('Debe hacer render el componente correctamente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });
    test('debe mostrar una lista de categorias', () => {
        const categories = ['goku', 'pokemon', 'hola'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(3);
    })


})
