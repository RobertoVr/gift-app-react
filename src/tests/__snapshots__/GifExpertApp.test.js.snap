// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente <GifExpertApp /> Debe hacer render el componente correctamente 1`] = `
<div>
  <h2>
    GifExpert
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`Probando componente <GifExpertApp /> debe mostrar una lista de categorias 1`] = `
<div>
  <h2>
    GifExpert
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="goku"
      key="goku"
    />
    <GifGrid
      category="pokemon"
      key="pokemon"
    />
    <GifGrid
      category="hola"
      key="hola"
    />
  </ol>
</div>
`;
